controllers/clusterversion_controller.go

```diff
@@ -604,6 +624,63 @@ func labelClientOperatorSubscription(c *ClusterVersionReconciler) error {
 		}
 	}
 
+	c.subscriptionChannel = sub.Spec.Channel
+
 	c.log.Info("successfully labelled ocs-client-operator subscription")
 	return nil
 }
+
+func (c *ClusterVersionReconciler) reconcileSubscription() error {
+
+	storageClients := &v1alpha1.StorageClientList{}
+	if err := c.list(storageClients); err != nil {
+		return fmt.Errorf("failed to list storageclients: %v", err)
+	}
+
+	var desiredChannel string
+	for idx := range storageClients.Items {
+		storageClient := &storageClients.Items[idx]
+		annotations := storageClient.GetAnnotations()
+		if annotatedChannel, ok := annotations[utils.DesiredSubscriptionChannelAnnotationKey]; ok {
+			if desiredChannel != "" && desiredChannel != annotatedChannel {
+				// not all storageclients desiring to be on same channel
+				return nil
+			}
+			desiredChannel = annotatedChannel
+		}
```

Comment:
The intent here is to skip empty annotation and extract a single value that meets all non-empty annotations. They way the code is written does not reflect the intent:

```suggestion
		// guarantee to return a string even if annotations is a nil map
		channel := storageClients.
			Items[idx].
			GetAnnotations()[utils.DesiredSubscriptionChannelAnnotationKey]

		// Skip clients with no/empty desired channel annotation
		if channel != "" {
			// Check if we already established a desired channel
			if desiredChannel == "" {
				desiredChannel = channel
			}

			// Check for agreement between clients	
			if channel != desiredChannel {
					desiredChannel = ""
					break
				} 
			}
		}
```